pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -237,18 +237,24 @@ func (cm *certificateManager) getKeyPair(cli client.Client, secretName, secretNa
 		return nil, nil, err
 	}
 
-	if x509Cert.NotAfter.Before(time.Now()) || x509Cert.NotBefore.After(time.Now()) {
+	timeInvalid := x509Cert.NotAfter.Before(time.Now()) || x509Cert.NotBefore.After(time.Now())
+	if timeInvalid || !ValidForClientAndServer(x509Cert) {
 		if !readCertOnly && strings.HasPrefix(x509Cert.Issuer.CommonName, rmeta.TigeraOperatorCAIssuerPrefix) {
 			if cm.keyPair.CertificateManagement != nil {
 				// When certificate management is enabled, we can simply return a certificate management key pair;
 				// the old secret will be deleted automatically.
 				return certificateManagementKeyPair(cm, secretName, nil), nil, nil
 			}
+
 			// We return nil, so a new secret will be created for expired (legacy) operator signed secrets.
 			return nil, nil, nil
 		}
+
 		// We return an error for byo secrets.
-		return nil, nil, fmt.Errorf("secret %s is not valid at this date", secretName)
+		if timeInvalid {
+			return nil, nil, fmt.Errorf("secret %s/%s is not valid at this date", secretNamespace, secretName)
+		}
+		return nil, nil, fmt.Errorf("secret %s/%s must specify ExtKeyusageClientAuth", secretNamespace, secretName)
```

Comment:
This means we won't ever get this secret recreated since we're returning an error here, at least that is my understanding from previous comments on this PR.
Also I tried these changes in cluster and the operator kept logging `secret <ns>/<name> must specify ExtKeyusageClientAuth` for several secrets.
Should we create an error type that we can return here and then check for that error type in the caller and handle the error as another case for when we need to create the secret?

Reply:
Oh I see we shouldn't get here because of the line `if timeInvalid || !ValidForClientAndServer(x509Cert) {` should handle the case I'm concerned with.